packages/vite/src/plugins/composable-keys.ts

```diff
@@ -116,7 +116,8 @@ export const composableKeysPlugin = createUnplugin((options: ComposableKeysOptio
           }
 
           // TODO: Optimize me (https://github.com/nuxt/framework/pull/8529)
-          const endsWithComma = code.slice(codeIndex + (node as any).start, codeIndex + (node as any).end - 1).trim().endsWith(',')
+          const newCode = code.slice(codeIndex + (node as any).start, codeIndex + (node as any).end - 1).trim()
```

Comment:
@danielroe this entire slice/trim operation, is it even needed? Can we not just
```ts
code[codeIndex + (node as any).end - 1]===','
```

Reply:
yes, I think it's possible it will end with whitespace...

Reply:
Hmm, I'll try to run some tests with this. There's gotta be a way to make it work without it..

Reply:
you could iterate backwards in a while loop... 🤔

Reply:
That could be possible, it will at the very least reduce the amount of characters "digested" by trim I suppose, but I wonder if there's a way to make it even faster, like if the expression itself will not contain whitespace, in which case last index could be used directly.

Reply:
![image](https://github.com/nuxt/nuxt/assets/92037085/ea8b358f-ba82-4e40-9357-5eb32fab3d63)

For starter optimization, we can use trimEnd which increase performance by ~50%